src/components/ItemCard/ItemCard.jsx

```diff
@@ -0,0 +1,89 @@
+import { useState, useEffect } from 'react';
+import { getItemList, itemList } from '../../features/slices/item';
+import { useSelector, useDispatch } from '../../app/store';
+
+function ItemCard() {
+  const itemData = useSelector(itemList);
+  const [items, setItems] = useState([]);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(getItemList(itemData));
+    setItems(itemData);
+  }, [items]);
+  return (
```

Comment:
Doesn't this make the component rerender so many times?!!

Because you added items as a dependency and you set the items inside the useeffect, when the useeffect run the items will be change and it will call useeffect again and again

Reply:
Thanks ayub! not really, no. it console.logged only once when i had a console.log

Reply:
Did you put the console log in this version? I mean with all this dispatch and setItems ?